client/extensions/woocommerce/state/sites/settings/general/handlers.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Internal dependencies
+ */
+import { areSettingsGeneralLoaded } from 'woocommerce/state/sites/settings/general/selectors';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { setError } from '../../status/wc-api/actions';
+import {
+	WOOCOMMERCE_SETTINGS_GENERAL_REQUEST,
+	WOOCOMMERCE_SETTINGS_GENERAL_REQUEST_SUCCESS,
+} from 'woocommerce/state/action-types';
+
+export const handleSettingsGeneralSuccess = ( { dispatch }, action, next, { data } ) => {
+	const { siteId } = action;
+	dispatch( {
+		type: WOOCOMMERCE_SETTINGS_GENERAL_REQUEST_SUCCESS,
```

Comment:
Okay. I went back to dispatching `WOOCOMMERCE_SETTINGS_GENERAL_REQUEST_FAILURE` which is what we had before and is consistent with other requests we have right now. If we decide to do any changes around action names, we can do that separate from moving to the data layer.